src/proxy.rs

```diff
@@ -180,14 +181,17 @@ impl Service for Proxy {
     }
 }
 
-pub fn listen(addr: SocketAddr, pool: Pool) -> result::Result<thread::JoinHandle<()>, io::Error> {
+pub fn listen(addr: SocketAddr, pool: Pool) -> result::Result<(thread::JoinHandle<()>, u16), io::Error> {
+    let (tx, rx) = channel();
     let handle = thread::spawn(move || {
 
         let mut core = Core::new().unwrap();
         let handle = core.handle();
 
         let listener = TcpListener::bind(&addr, &handle).unwrap();
-        info!("Listening on http://{}", &addr);
+        let local_addr = listener.local_addr().unwrap();
```

Comment:
the code is inside the `thread::spawn`. Difficult to use a `try!` here.
Do you have an idea?